sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-processing.service.ts

```diff
@@ -0,0 +1,188 @@
+        documentNumbers,
+      );
+    const disbursementScheduleMap = {};
+    disbursementSchedules.forEach(
+      (disbursementSchedule) =>
+        (disbursementScheduleMap[disbursementSchedule.documentNumber] =
+          disbursementSchedule.id),
+    );
+
+    this.logger.log(
+      "Processing all the records in file with valid data and document number that belongs to SIMS.",
+    );
+
+    const createdAt = new Date();
+    for (const response of responseData.records) {
+      const disbursementScheduleId =
+        disbursementScheduleMap[response.documentNumber];
+      if (disbursementScheduleId) {
+        const disbursementReceipt = new DisbursementReceipt();
+        disbursementReceipt.batchRunDate = responseData.header.batchRunDate;
+        disbursementReceipt.studentSIN = response.studentSIN;
+        disbursementReceipt.disbursementSchedule = {
+          id: disbursementScheduleId,
+        } as DisbursementSchedule;
+        disbursementReceipt.fundingType = response.fundingType;
+        disbursementReceipt.totalEntitledDisbursedAmount =
+          response.totalEntitledDisbursedAmount;
+        disbursementReceipt.totalDisbursedAmount =
+          response.totalDisbursedAmount;
+        disbursementReceipt.disburseDate = response.disburseDate;
+        disbursementReceipt.disburseAmountStudent =
+          response.disburseAmountStudent;
+        disbursementReceipt.disburseAmountInstitution =
+          response.disburseAmountInstitution;
+        disbursementReceipt.dateSignedInstitution =
+          response.dateSignedInstitution;
+        disbursementReceipt.institutionCode = response.institutionCode;
+        disbursementReceipt.disburseMethodStudent =
+          response.disburseMethodStudent;
+        disbursementReceipt.studyPeriodEndDate = response.studyPeriodEndDate;
+        disbursementReceipt.totalEntitledGrantAmount =
+          response.totalEntitledGrantAmount;
+        disbursementReceipt.totalDisbursedGrantAmount =
+          response.totalDisbursedGrantAmount;
+        disbursementReceipt.totalDisbursedGrantAmountStudent =
+          response.totalDisbursedGrantAmountStudent;
+        disbursementReceipt.totalDisbursedGrantAmountInstitution =
+          response.totalDisbursedGrantAmountInstitution;
+        disbursementReceipt.creator = { id: auditUserId } as User;
+        disbursementReceipt.createdAt = createdAt;
+        disbursementReceipt.disbursementReceiptValues = response.grants.map(
+          (grant) => {
+            const disbursementReceiptValue = new DisbursementReceiptValue();
+            disbursementReceiptValue.grantType = grant.grantType;
+            disbursementReceiptValue.grantAmount = grant.grantAmount;
+            disbursementReceiptValue.creator = { id: auditUserId } as User;
+            disbursementReceiptValue.createdAt = createdAt;
+            return disbursementReceiptValue;
+          },
+        );
+        try {
```

Comment:
IMO, the `try` should involve more code to ensure that even small issues will not stop the processing.